core/execution/engine.go

```diff
@@ -710,26 +911,41 @@ func (e *Engine) SubmitStopOrders(
 		market = submission.RisesAbove.OrderSubmission.MarketID
 	}
 
-	mkt, ok := e.markets[market]
-	if !ok {
-		return nil, types.ErrInvalidMarketID
-	}
+	if mkt, ok := e.markets[market]; ok {
```

Comment:
Not a big deal, but there's a few method where the same code is duplicated for both maps of markets if not mistaken. I would have probably done something like:
```
type CommonMarket interface {
   SubmitOrder(...)
   CancelOrder(...)
   ... 
}

func (e *Engine) getMarket(id string) CommonMarket {
   if m := e.markets[id]; m != nil {
      return m
   }

   return e.spotMatkets[id]
}  
```

Reply:
actual looking at the rest of the file, it would seem that we do that on pretty much all method, seems like a source of future bug where we'd change things for one type of market and not the other

Reply:
it's true for many but not all cases as there's a few places related to successor market where the future call has more/different logic than the spot one. still worth doing.